agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/DatabaseClientTracer.java

```diff
@@ -87,17 +117,14 @@ protected Span afterStart(final Span span) {
   }
 
   /** This should be called when the connection is being used, not when it's created. */
-  public Span onConnection(final Span span, final CONNECTION connection) {
-    assert span != null;
-    if (connection != null) {
-      span.setAttribute(SemanticAttributes.DB_USER.key(), dbUser(connection));
-      span.setAttribute(SemanticAttributes.DB_INSTANCE.key(), dbInstance(connection));
-      span.setAttribute(SemanticAttributes.DB_URL.key(), dbUrl(connection));
-    }
+  protected Span onConnection(final Span span, final CONNECTION connection) {
```

Comment:
Just a note not necessarily this PR but I was confused by having both `onConnection` and `onPeerConnection`, I think a connection always has a peer. Based on the javadoc, maybe it's `onConnectionAttempt` and `onConnectionComplete` but not quite sure

Reply:
I am always confused by that as well. There will be a round of cleanup of all new tracers, when all/majority of instrumentations are migrated.